src/test/unit_tests/jest_tests/test_config_dir_restructure_upgrade_script.test.js

```diff
@@ -0,0 +1,654 @@
+            actual_upgrade_params = await prepare_account_upgrade_params(default_config_fs, new_account_data.name);
+            assert_upgrade_params(actual_upgrade_params, expected_account_upgrade_params);
+        } catch (err) {
+            fail(err);
+        } finally {
+            if (src_file) await src_file.close(default_config_fs.fs_context);
+            if (actual_upgrade_params.src_file) await actual_upgrade_params.src_file.close(default_config_fs.fs_context);
+        }
+    });
+
+    it('prepare_account_upgrade_params() - account name exists', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
+        await expect(prepare_account_upgrade_params(default_config_fs, new_account_data.name)).rejects.toThrow('No such file or directory');
+    });
+});
+
+describe('upgrade_account_config_file', () => {
+    beforeAll(async () => {
+        await fail_test_if_default_config_dir_exists('test_config_dir_restructure', default_config_fs);
+    });
+
+    beforeEach(async () => {
+        await create_config_dir(config.NSFS_NC_DEFAULT_CONF_DIR);
+        await default_config_fs.create_config_dirs_if_missing();
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        await create_fresh_path(hidden_old_accounts_path);
+        await create_fresh_path(hidden_access_keys_backup_path);
+    }, TEST_TIMEOUT);
+
+    afterEach(async () => {
+        await clean_config_dir(default_config_fs);
+    }, TEST_TIMEOUT);
+
+    it('upgrade_account_config_file() - account exists in accounts/', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data, { symlink_access_key: true });
+        await upgrade_account_config_file(default_config_fs, new_account_data.name, hidden_access_keys_backup_path, dbg);
+        await assert_account_config_file_upgraded({[new_account_data.name]: new_account_data});
+    });
+
+    it('upgrade_account_config_file() - account was already upgraded', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data, { symlink_access_key: false });
+        await write_manual_config_file(CONFIG_TYPES.ACCOUNT, default_config_fs, new_account_data, undefined,
+            { symlink_access_key: false, symlink_name: false });
+        await upgrade_account_config_file(default_config_fs, new_account_data.name, hidden_access_keys_backup_path, dbg);
+        await assert_account_config_file_upgraded({[new_account_data.name]: new_account_data});
+    });
+
+    it('upgrade_account_config_file() - identity exists but indexes are not', async () => {
```

Comment:
What do you mean by "identity exists but indexes are not"? (the term "indexes" is links?)
Could you write the title with what explicitly is missing (access key link, identity link, etc.).